src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetriesBehavior.cs

```diff
@@ -28,28 +29,26 @@ public FirstLevelRetriesBehavior(FlrStatusStorage storage, FirstLevelRetryPolicy
                 var messageId = context.Message.MessageId;
                 var pipelineUniqueMessageId = messageId;
 
-                var numberOfFailures = storage.GetFailuresForMessage(pipelineUniqueMessageId);
+                storage.RecordFailedProcessingAttempt(pipelineUniqueMessageId, ExceptionDispatchInfo.Capture(ex));
```

Comment:
you probably missed my comment since it's now shown due to your renaming. I still think this should go after line 40.
 Please have a look at this comment: https://github.com/Particular/NServiceBus/pull/3614#discussion_r58693790